sdk/language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -145,8 +145,17 @@ main = do
                            PkgId _ -> unPackageId pkgId
                            PkgNameVer (pkgName, _) -> unPackageName pkgName <> " (hash: " <> unPackageId pkgId <> ")"
                          pkgUnitId = maybe (unPackageId pkgId) unitIdToText $ packageUnitId pkg
-                     T.putStrLn $ "Generating " <> pkgDesc
-                     daml2js Daml2jsParams{..}
+                         isUtilityPackage =
+                          all (\mod ->
+                            null (moduleTemplates mod)
+                              && null (moduleInterfaces mod)
+                              && not (any (getIsSerializable . dataSerializable) $ moduleDataTypes mod)
+                          ) $ packageModules pkg
+                     if isUtilityPackage
+                       then T.putStrLn $ "Skipping " <> pkgDesc <> " as it does not define any serializable types"
+                       else do
+                         T.putStrLn $ "Generating " <> pkgDesc
+                         daml2js Daml2jsParams{..}
```

Comment:
The check is only needed at the top level, as ghc already enforces that dependencies cannot have the same unit id.